sources/packages/backend/apps/api/src/institution-integration/ier-integration/ier12-file.service.ts

```diff
@@ -0,0 +1,172 @@
+    const pendingAssessments =
+      await this.studentAssessmentService.getPendingAssessment(generatedDate);
+    if (!pendingAssessments.length) {
+      return [
+        {
+          generatedFile: "none",
+          uploadedRecords: 0,
+        },
+      ];
+    }
+    this.logger.log(`Found ${pendingAssessments.length} assessments.`);
+    const fileRecords: Record<string, IER12Record[]> = {};
+    pendingAssessments.forEach((pendingAssessment) => {
+      const institutionCode =
+        pendingAssessment.offering.institutionLocation.institutionCode;
+      if (!fileRecords[institutionCode]) {
+        fileRecords[institutionCode] = [];
+      }
+      fileRecords[institutionCode].push(
+        this.createIER12Record(pendingAssessment),
+      );
+    });
+    const uploadResult: IER12UploadResult[] = [];
+    try {
+      this.logger.log("Creating IER 12 request content...");
+      for await (const ierUploadResult of Object.keys(fileRecords).map(
+        (institutionCode) =>
+          this.uploadIER12Content(institutionCode, fileRecords),
+      )) {
+        uploadResult.push(ierUploadResult);
+      }
+    } catch (error) {
+      this.logger.error(
+        `Error while uploading content for IER 12 Request: ${error}`,
+      );
+      throw error;
+    }
+    return uploadResult;
+  }
+
+  /**
+   * Upload the content in SFTP server location.
+   * @param institutionCode Institution code for the file generated.
+   * @param fileRecords Total records with institutionCode.
+   * @returns Updated records count with filepath.
+   */
+  async uploadIER12Content(
+    institutionCode: string,
+    fileRecords: Record<string, IER12Record[]>,
+  ): Promise<IER12UploadResult> {
+    try {
+      // Create the Request content for the IER 12 file by populating the content.
+      const fileContent = this.ierIntegrationService.createIER12FileContent(
+        fileRecords[institutionCode],
+      );
+      // Create the request filename with the file path for the each and every institutionCode.
+      const fileInfo = this.createRequestFileName(institutionCode);
+      this.logger.log("Uploading content...");
+      await this.ierIntegrationService.uploadContent(
+        fileContent,
+        fileInfo.filePath,
+      );
+      this.logger.log("Content uploaded.");
+      return {
+        generatedFile: fileInfo.filePath,
+        uploadedRecords: fileContent.length,
+      };
+    } catch (error) {
+      this.logger.error(
+        `Error while uploading content for IER 12 file: ${error}`,
+      );
+      throw error;
+    }
+  }
+
+  /**
+   * Create file name IER 12 records file.
+   * @param institutionCode institutionCode to be a part of the filename.
+   * @returns Full file path of the file to be saved on the SFTP.
+   */
+  createRequestFileName(institutionCode: string): {
+    fileName: string;
+    filePath: string;
+  } {
+    const timestamp = getFileNameAsCurrentTimestamp();
+    const fileName = `IER_012_${timestamp}.txt`;
```

Comment:
`IER_12_` would be probably enough, right?

Reply:
<img width="852" alt="image" src="https://user-images.githubusercontent.com/62901416/201711789-ffb8ba8c-f3e5-4a23-9790-1bbb6e299cb1.png">

<img width="890" alt="image" src="https://user-images.githubusercontent.com/62901416/201711541-cff3cd81-a624-4725-b68a-048f0013eb52.png">
 There is nothing given in the document for IER12, but for IER10 and IER11 it was given as IER_010 and IER_011. So just followed it for IER12 types too.